frontend/app/components/Filter/FilterGroup.tsx

```diff
@@ -0,0 +1,66 @@
+'use client'
+
+import {GetCountriesQueryResult, GetGenresQueryResult} from '@/sanity.types'
+import {ChevronDown, ChevronUp} from 'lucide-react'
+import {useState} from 'react'
+import {useSearchParams} from 'next/navigation'
+
+export default function FilterGroup({
+  category,
+  initCollapse,
+}: {
+  category: {
+    label: string
+    filters: GetGenresQueryResult | GetCountriesQueryResult
+  }
+  initCollapse?: boolean
+}) {
+  const searchParams = useSearchParams()
+  const [isCollapsed, setIsCollapsed] = useState(searchParams.has(category.label) || initCollapse)
+  const handleCollapse = () => setIsCollapsed(!isCollapsed)
+  return (
+    <fieldset
+      id={category.label}
+      name={category.label}
+      className="border-b border-black/25 py-4 flex flex-col gap-2"
+    >
+      <div
+        className="flex items-center justify-between hover:cursor-pointer"
+        role="button"
```

Comment:
I would be careful using role="button". 
This is because when you use the actual <button> component. You get a lot of good stuff out of the box.